### PR TITLE
Add new HCPVaultSecretsApp CRD and Controller

### PR DESCRIPTION
**Pull request by benashz**
Extends VSO to support syncing HCP Vault Secrets to K8s.

Sample HCPVaultSecretsApp Custom Resource:
```yaml
apiVersion: secrets.hashicorp.com/v1beta1
kind: HCPVaultSecretsApp
metadata:
  name: vso-db-demo-hcp-vsa
  namespace: demo-ns
spec:
  appName: vso
  destination:
    create: true
    name: vso-db-demo-hcp-vsa
```

### CODE REVIEW

File: internal/common/common.go [hunk 1]
```diff
@@ -25,11 +25,25 @@ import (
 var (
 	// OperatorNamespace of the current operator instance, set in init()
 	OperatorNamespace                   string
-	InvalidObjectKeyError               = fmt.Errorf("invalid objectKey")
-	InvalidObjectKeyErrorEmptyName      = fmt.Errorf("%w, empty name", InvalidObjectKeyError)
-	InvalidObjectKeyErrorEmptyNamespace = fmt.Errorf("%w, empty namespace", InvalidObjectKeyError)
+	InvalidObjectKeyError                      = fmt.Errorf("invalid objectKey")
+	InvalidObjectKeyErrorEmptyName             = fmt.Errorf("%w, empty name", InvalidObjectKeyError)
+	InvalidObjectKeyErrorEmptyNamespace        = fmt.Errorf("%w, empty namespace", InvalidObjectKeyError)
+	defaultMaxRetries                   uint64 = 60
```
Comment (kschoche) [inline internal/common/common.go:31]: nit: Is it me, or does something about the alignment feel off? I guess everything is aligned on `=`, but what about `OperatorNamespace`? Anyways, such a minor thing anyway.

File: internal/common/common.go [hunk 2]
```diff
@@ -25,11 +25,25 @@ import (
...
+	defaultRetryDuration                       = time.Millisecond * 500
 )
 
+type NamespaceNotAllowedError struct {
+	TargetNS  string
+	ObjRef    types.NamespacedName
+	AllowedNS []string
+}
+
+func (n *NamespaceNotAllowedError) Error() string {
```
Comment (kschoche) [inline internal/common/common.go:41]: Nice change!

**Review by kschoche — APPROVED** [review]
I think this looks great!
Excellent work on all of the refactoring and cleanup too.

**Review by tvoran — APPROVED** [review]
Just some small suggestions.

File: internal/common/common.go (hunk 1 above)
Comment (benashz) [reply internal/common/common.go:31]: Yeah, I think that's  just the way that `go fmt` does it?

File: controllers/hcpvaultsecretsapp_controller.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	if err != nil {
+		logger.Error(err, "Get App Secret", "appName", o.Spec.AppName)
+		return ctrl.Result{
+			RequeueAfter: computeHorizonWithJitter(requeueDurationOnError),
+		}, nil
+	}
+
+	data, err := r.SecretDataBuilder.WithHVSAppSecrets(resp)
+	if err != nil {
+		logger.Error(err, "MakeVaultSecretK8sData", "appName", o.Spec.AppName)
```
[the remark this answers is not in the document]
Comment (benashz) [reply controllers/hcpvaultsecretsapp_controller.go]: Fixed in 2b8d2093ac9436239283cc382493dda9f1e34ad8

File: internal/common/common.go (hunk 2 above)
Comment (benashz) [reply internal/common/common.go:41]: Thanks!

File: config/crd/patches/cainjection_in_hcpvaultsecretsapps.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
```
[the remark this answers is not in the document]
Comment (benashz) [reply config/crd/patches/cainjection_in_hcpvaultsecretsapps.yaml]: Done. Thanks!

**Pull request merged**